### PR TITLE
Document socket-proxy permissions, return early when update failed on scaling down

### LINKED ISSUE #342: unexpected behaviour with docker socket proxy
**Describe the bug**
Service downscaling seemingly hangs when proper rights are not set on the docker socket proxy.

**To Reproduce**
```yaml
...
services:
  target:
    ...
    deploy:
      labels:
        docker-volume-backup.stop-during-backup: sonarr

  minio_backup:
    image: offen/docker-volume-backup:v2.36.1
    environment:
      BACKUP_STOP_DURING_BACKUP_LABEL: anything
      DOCKER_HOST: tcp://docker_socket_proxy:2375
    ...
  docker_socket_proxy:
    image: tecnativa/docker-socket-proxy:0.1
    environment:
      CONTAINERS: 1
      SERVICES: 1
      ALLOW_START: 1
      ALLOW_STOP: 1
      INFO: 1
#      POST: 1
    ...
```
Forget adding `POST` permissions for the proxy.

**Expected behavior**
Script panics early due to lack of permissions. I believe we are stuck in this [wait](https://github.com/offen/docker-volume-backup/blob/main/cmd/backup/stop_restart.go#L223). I think if the scale down fails, we should not wait for it to complete. 

**Version (please complete the following information):**
 - Image Version: v2.36.1
 - Docker Version: 25.0.1, build 29cf629

**Additional context**
It would also be handy to provide a sample to the [docker-socket-proxy](https://offen.github.io/docker-volume-backup/how-tos/use-custom-docker-host.html) configuration with maybe comments on which permission is required for which setting. ( For example we early panic if `INFO` is not present. )

### PR DESCRIPTION
**Pull request by m90**
Closes #342

### CODE REVIEW

**Comment by m90** [comment]
@pixxon Maybe you can have a quick look at the table of permissions, as I'm not a docker-socket-proxy user myself, I might have missed something here and there, and a quick sanity check would be much appreciated.

**Comment by pixxon** [comment]
I am not entirely sure what is going on, but seemingly `ALLOW_STOP` and `ALLOW_START` are useless. Without the `POST` permission, they just fail. ( Makes sense since it is a POST request. ) However the `CONTAINERS` allows stop and start anyway.

Related stop and start requests:
```
dockerfrontend dockerbackend/dockersocket 0/0/0/10305/10305 204 176 - - ---- 1/1/0/0/0 0/0 "POST /v1.43/containers/b8f0c74331b45c4242d0c72375ffa605278700a8d38cf4622d31cb1c06ccf6ae/stop HTTP/1.1"
dockerfrontend dockerbackend/dockersocket 0/0/0/240/240 204 176 - - ---- 1/1/0/0/0 0/0 "POST /v1.43/containers/b8f0c74331b45c4242d0c72375ffa605278700a8d38cf4622d31cb1c06ccf6ae/start HTTP/1.1"
```

File: docs/how-tos/use-custom-docker-host.md [hunk 1]
```diff
@@ -13,5 +13,36 @@ If you are interfacing with Docker via TCP, set `DOCKER_HOST` to the correct URL
...
+
+---
+
+In case you are using [`docker-socket-proxy`][proxy], this means following permissions are required:
+
+| Permission | When |
+|-|-|
+| INFO | always required |
+| CONTAINERS | always required |
+| POST | required when using `stop-during-backup` labels |
```
Comment (pixxon) [inline docs/how-tos/use-custom-docker-host.md:39]: required for `exec` labeled commands too

File: docs/how-tos/use-custom-docker-host.md [hunk 2]
```diff
@@ -13,5 +13,36 @@ If you are interfacing with Docker via TCP, set `DOCKER_HOST` to the correct URL
...
+| EXEC | required when using `exec`-labeled commands |
+| SERVICES | required when running in Swarm mode |
+| NODES | required when using `stop-during-backup` and running in Swarm mode |
+| TASKS | required when using `stop-during-backup` and running in Swarm mode |
```
Comment (pixxon) [inline docs/how-tos/use-custom-docker-host.md:43]:
```suggestion
| NODES | required when labeling services `stop-during-backup` |
| TASKS | required when labeling services `stop-during-backup` |
```

Staying consistent with the one below and services imply the Swarm mode.

**Comment by m90** [comment]
Thanks :tophat:

**Pull request merged**